libraries/app/database_api.cpp

```diff
@@ -226,6 +226,61 @@ class database_api_impl : public std::enable_shared_from_this<database_api_impl>
          FC_ASSERT( account, "no such account" );
          return account;
       }
+      market_ticker buildTicker(const market_ticker_object& mto) const
+      {
+         const auto assets = get_assets( { mto.base, mto.quote } );
+         const fc::time_point_sec now = _db.head_block_time();
+
+         market_ticker ticker;
+         ticker.time = now;
+         ticker.base = assets[0]->symbol;
+         ticker.quote = assets[1]->symbol;
+         ticker.latest = "0";
+         ticker.lowest_ask = "0";
+         ticker.highest_bid = "0";
+         ticker.percent_change = "0";
```

Comment:
Code here constructs a default object then modify it. I prefer using a constructor with parameters since IMHO it has better performance.